### PR TITLE
RocketMQ gets the consumer group

### PR DESCRIPTION
**Pull request by xiaoyao2246**
I want to get the RocketMq consumer group，So I changed the code。Please help me to see if this is right?

### CODE REVIEW

**Comment by laurit** [comment]
@xiaoyao2246 please change target branch of the pull request to `main`

File: instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java [hunk 1]
```diff
@@ -28,7 +29,12 @@ public void consumeMessageBefore(ConsumeMessageContext context) {
       return;
     }
     Context parentContext = Context.current();
-    Context newContext = instrumenter.start(parentContext, context.getMsgList());
+
+    String consumerGroup = context.getConsumerGroup();
+    Baggage baggage = Baggage.fromContext(parentContext);
+    Baggage newBaggage = baggage.toBuilder().put("consumerGroup", consumerGroup).build();
```
Comment (laurit) [inline instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java:35]: Using baggage to pass info is really roundabout, you could change the type of the request that is passed to `instrumenter.start` to an object that contains both message list and consumer group. For example see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c166c937b3417dbc6bfcaafd7321483fcf0c0c37/instrumentation/spring/spring-integration-4.1/library/src/main/java/io/opentelemetry/instrumentation/spring/integration/TracingChannelInterceptor.java#L73 message and message channel are packed into `MessageWithChannel` that is used as request.

File: instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/RockerMqConsumerAttributeExtractor.java [hunk 2]
```diff
@@ -24,7 +23,7 @@ protected String system(MessageExt request) {
 
   @Override
   protected String destinationKind(MessageExt request) {
-    return SemanticAttributes.MessagingDestinationKindValues.TOPIC;
+    return "group";
```
Comment (laurit) [inline instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/RockerMqConsumerAttributeExtractor.java:26]: This method is supposed to return values defined in specification, which are `topic` and `queue`. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md If you want to introduce new values you should first change the specification.

Comment (xiaoyao2246) [reply instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/RockerMqConsumerAttributeExtractor.java:26]: You mean to add group to constant class?

File: instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java (hunk 1 above)
Comment (xiaoyao2246) [reply instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java:35]: Are you talking about adding a custom RocketMQ interceptor to implement the functionality in the same way that Spring implements interceptors?

Comment (laurit) [reply instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java:35]: No, I meant that you could create a class `RocketMqRequest` that contains a `MessageExt` and consumer group. In `RocketMqConsumerInstrumenter` change the instrumenters that are parametrized over `MessageExt` to be parametrized over `RocketMqRequest`. This way you should be able to get consumer group in `RocketMqConsumerAttributeGetter` or `RocketMqConsumerExperimentalAttributeExtractor`.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/messaging/MessagingAttributesExtractor.java [hunk 3]
```diff
@@ -33,7 +34,15 @@ public final void onStart(AttributesBuilder attributes, Context parentContext, R
       set(attributes, SemanticAttributes.MESSAGING_TEMP_DESTINATION, true);
       set(attributes, SemanticAttributes.MESSAGING_DESTINATION, TEMP_DESTINATION_NAME);
     } else {
-      set(attributes, SemanticAttributes.MESSAGING_DESTINATION, destination(request));
+      if (request.getClass().getName().equals("org.apache.rocketmq.common.message.MessageClientExt")) {
```
Comment (laurit) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/messaging/MessagingAttributesExtractor.java:37]: You'll have to find some other way of doing this. You should not add any rocketmq specific stuff to the common code.

File: instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/RockerMqConsumerAttributeExtractor.java (hunk 2 above)
Comment (laurit) [reply instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/RockerMqConsumerAttributeExtractor.java:26]: To add a new constant you'll have to add it to specification. I'd suggest you to consider whether you really need to change this.

File: instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java (hunk 1 above)
Comment (xiaoyao2246) [reply instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java:35]: OK，Let me try，Thank you。

**Pull request closed without merging**